identify/extensions.py

```diff
@@ -32,6 +33,7 @@
     'cxx': {'text', 'c++'},
     'dart': {'text', 'dart'},
     'def': {'text', 'def'},
+    'dll': {'binary'},
```

Comment:
I'm fine adding `dll`, but `.bin` is too generic of an extension (and doesn't necessarily mean it contains binary data)

you can also use `exclude: \.bin$` to exclude it.  if identify is identifying this as text it's likely that it doesn't contain any unprintable bytes?

Reply:
They do actually contain printable bytes, but are still to be treated as binary, being specifically bit test vectors for binary protocols. ( i.e. end-of-line-fixer and a number of other generic pre-commit hooks shouldn't touch these )

Adding an exclude to every hook is a workaround, but does make the config file quite a bit more verbose

Reply:
do you know about [global exclude](https://pre-commit.com/#top_level-exclude) ?

Reply:
I do, but there are other hooks that do need to run on those, so either way there's extra configuration that could be solved by convention.